src/webgpu/api/operation/shader_module/compilation_info.spec.ts

```diff
@@ -0,0 +1,156 @@
+    // Expected Error: vec4 should be vec4<f32>
+    return vec4(0.0, 0.0, 0.0, 1.0);
+  }
+`;
+
+g.test('compilationInfo_valid_shader')
+  .desc(
+    `Ensures that compilationInfo() can be called on a valid ShaderModules.`
+  )
+  .fn(async t => {
+    const shaderModule = t.device.createShaderModule({
+      code: `
+        [[stage(vertex)]] fn main() -> [[builtin(position)]] vec4<f32> {
+          return vec4<f32>(0.0, 0.0, 0.0, 1.0);
+        }
+      `,
+    });
+
+    const info = await shaderModule.compilationInfo();
+
+    t.expect(info instanceof GPUCompilationInfo, "Expected a GPUCompilationInfo object to be returned");
+
+    // Expect that we get zero error messages from a valid shader.
+    // Message types other than errors are OK.
+    let errorCount = 0;
+    for (const message of info.messages) {
+      if (message.type == 'error') {
+        errorCount++;
+      }
+    }
+    t.expect(errorCount == 0, "Expected zero GPUCompilationMessages of type 'error'");
+});
+
+g.test('compilationInfo_invalid_shader')
+  .desc(
+    `Ensures that compilationInfo() can be called on an invalid ShaderModules.`
+  )
+  .fn(async t => {
+
+    const shaderModule = t.expectGPUError(
+      'validation', // Should be a validation error since the shader is invalid.
+      () =>
+        t.device.createShaderModule({
+          code: BAD_SHADER_SOURCE,
+        })
+    );
+
+    const info = await shaderModule.compilationInfo();
+
+    t.expect(info instanceof GPUCompilationInfo, "Expected a GPUCompilationInfo object to be returned");
+
+    // Expect that we get at least one error message.
+    // Additional errors or other message types are also OK.
+    let errorCount = 0;
+    for (const message of info.messages) {
+      if (message.type == 'error') {
+        errorCount++;
+      }
+    }
+    t.expect(errorCount > 0, "Expected at least one GPUCompilationMessages of type 'error'");
+});
+
+g.test('compilationInfo_line_position')
+  .desc(
+    `Ensures that line numbers reported by compilationInfo either point at an appropriate line and
+    position or at 0:0, indicating an unknown position.`
+  )
+  .fn(async t => {
+
+    const shaderModule = t.expectGPUError(
+      'validation', // Should be a validation error since the shader is invalid.
+      () =>
+        t.device.createShaderModule({
+          code: BAD_SHADER_SOURCE,
+        })
+    );
+
+    const info = await shaderModule.compilationInfo();
+
+    let foundAppropriateError = false;
+    for (const message of info.messages) {
+      if (message.type == 'error') {
+        // Some backends may not be able to indicate a precise location for the error. In those
+        // cases a line and position of 0 should be reported.
+        if (message.lineNum == 0) {
+          foundAppropriateError = true;
+          t.expect(message.linePos == 0, "GPUCompilationMessages that don't report a line number should not report a line position.")

```

Comment:
nit: Test the converse as well, e.g.:
```ts
t.expect((message.lineNum === 0) === (message.lineNum === 0));
```
(outside of the `if (message.lineNum == 0)`)